scripts/referrer-user-count.ts

```diff
@@ -0,0 +1,105 @@
+import yargs from 'yargs'
+import { protocolFilters, supportedNetworkIds } from './consts'
+import { fetchReferralEvents, removeDuplicates } from './referrals'
+import { NetworkId, Protocol, protocols } from './types'
+import { writeFileSync } from 'fs'
+
+async function getArgs() {
+  const argv = await yargs
+    .env('')
+    .option('protocol', {
+      description: 'protocol that the referrals are for',
+      demandOption: true,
+      choices: protocols,
+      type: 'string',
+    })
+    .option('output-file', {
+      alias: 'o',
+      description: 'output file path to write JSON results',
+      type: 'string',
+      demandOption: true,
+    })
+    // This is coerced as a string array to prevent 0x... from being interpreted as a number
+    .option('referrer-addresses', {
```

Comment:
A referrer is not an address, but a unique referrer identifier (just an arbitrary string). For example `"valora"`, or a hypothetical frontend to Beefy, `"leefy"`, or something of the sort.